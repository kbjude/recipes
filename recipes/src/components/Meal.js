import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Meal = ({
  id, name, image, path,
}) => (
  <Link to={path}>
    <div>
      <h5>{id}</h5>
      <h3>{name}</h3>
      <img src={image} alt="" />
    </div>
  </Link>
);

Meal.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};

export default Meal;
