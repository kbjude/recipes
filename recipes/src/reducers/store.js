import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '.';

const middlewares = [thunk];

const store = createStore(rootReducer, applyMiddleware(...middlewares));
export default store;
