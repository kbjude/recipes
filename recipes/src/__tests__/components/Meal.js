import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import Meal from '../../components/Meal';

const character = {
  id: '2',
  name: 'Chicken',
  description: 'Chicken well cooked',
  image: 'www.mockurl.com',
};

describe('Rendering component', () => {
  it('creates a Meal component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Meal
              id={character.id}
              name={character.name}
              description={character.description}
              image={character.image}
              path="http://localhost:3000/meals/details/52874"
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Chicken');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Meal
            id={character.id}
            name={character.name}
            description={character.description}
            image={character.image}
            path="http://localhost:3000/meals/details/52874"
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
