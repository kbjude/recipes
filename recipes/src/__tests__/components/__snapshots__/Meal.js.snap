// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<a
  href="/http://localhost:3000/meals/details/52874"
  onClick={[Function]}
>
  <div>
    <h5>
      2
    </h5>
    <h3>
      Chicken
    </h3>
    <img
      alt=""
      src="www.mockurl.com"
    />
  </div>
</a>
`;
